app.py

```diff
@@ -101,9 +111,115 @@ def display_episode() -> None:
         messages = render_for_humans(episode)
         streamlit_rendering(messages)
 
+def get_unique_id():
+    """Generate or retrieve a unique identifier for the user."""
+    if "user_id" not in st.session_state:
+        # Generate a unique identifier for the user
+        st.session_state["user_id"] = str(uuid.uuid4())
+    return st.session_state["user_id"]
+
+def store_survey_response(survey_id, user_id, response_data):
```

Comment:
see the comment here (https://github.com/XuhuiZhou/HAICosystem-demo-internal/commit/50582e9c1d00a74c8de4479d9ff05dbf173a4e82)